src/interpreter/runtime/values/i32.js

```diff
@@ -1,14 +1,167 @@
 // @flow
-import { BaseNumber } from "./number";
+const Long = require("long");
+
+import { RuntimeError } from "../../../errors";
 
 const bits = 32;
 const type = "i32";
 
-export class i32 extends BaseNumber {}
+// the specification describes the conversion from unsigned to signed
+// https://webassembly.github.io/spec/core/exec/numerics.html#aux-signed
+// this function performs the inverse
+const toUnsigned = a => (a < 0 ? a + Math.pow(2, bits) : a);
 
-export function createValueFromAST(value: number): StackLocal {
-  value = (value | 0) % Math.pow(2, bits);
+export class i32 implements Number<i32> {
```

Comment:
You probably need to rebase, I had to rename the interface.

Reply:
Ah - good spot, I've turned Flow back on again, and fixed the issues this PR introduced :-)